src/main/java/seedu/address/logic/commands/DeleteTagCommand.java

```diff
@@ -54,14 +84,42 @@ public CommandResult execute(Model model) throws CommandException {
         return new CommandResult(successMessage + currentTags);
     }
 
+    /**
+     * Returns a set of tags that are in use by the persons in the address book.
+     * @param model
+     */
+    private Set<Tag> getTagsInUse(Model model) {
+        Set<Tag> tagsInUse = new HashSet<>();
+        List<Person> persons = model.getFullPersonList();
+        for (Person person : persons) {
+            Set<Tag> personTags = person.getTags();
+            for (Tag tag : tags) {
+                if (personTags.contains(tag)) {
+                    tagsInUse.add(tag);
+                }
+            }
+        }
+        return tagsInUse;
```

Comment:
Might have slight issue with 'arrowhead' method, could possibly extract out the inner for loop (tag) to be a private method. i.e addPersonTags(personTags, tagsInUse)